### PR TITLE
Add support to set ssl version

### PR DESCRIPTION
**Pull request by siyengar**
Adds support to set the SSL version in nifty. The default netty implementation does not provide this functionality. We could add this to netty however we'd have to upgrade to the latest netty version before we can do this. This gives us that functionality until then.

### CODE REVIEW

File: nifty-ssl/src/main/java/com/facebook/nifty/ssl/OpenSslServerConfiguration.java [hunk 1]
```diff
@@ -16,21 +16,25 @@
...
+        TLS1_2, // Server will accept only TLS1.2.
+    };
+
     public static class Builder extends SslServerConfiguration.BuilderBase<Builder> {
 
         public SessionTicketKey[] ticketKeys;
         // A string that can be used to separate tickets from different entities.
         public String sessionContext = "thrift";
         public long sessionTimeoutSeconds = 86400;
+        public SSLVersion sslVersion = SSLVersion.TLS;
```
Comment (alandau) [inline nifty-ssl/src/main/java/com/facebook/nifty/ssl/OpenSslServerConfiguration.java]: Should we default to 1.2?

File: nifty-ssl/src/main/java/com/facebook/nifty/ssl/NiftyOpenSslServerContext.java [hunk 2]
```diff
@@ -0,0 +1,318 @@
...
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * A lot of this code is taken from Netty's OpenSslServerContext.
+ * NiftyOpenSslServerContext allows us to control this instantiation of SSLContext with custom options
+ * missing in netty. We should be able to get rid of this when we move the dependency
+ * to the latest version of netty.
```
Comment (alandau) [inline nifty-ssl/src/main/java/com/facebook/nifty/ssl/NiftyOpenSslServerContext.java]: Please add a comment with the version number you copied it from. Just for future reference

File: nifty-core/src/main/java/com/facebook/nifty/ssl/SslHandlerFactory.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+ * limitations under the License.
+ */
+package com.facebook.nifty.ssl;
+
+import org.jboss.netty.handler.ssl.SslHandler;
+
+/**
+ * A Factory that returns a handler.
+ */
+public interface SslHandlerFactory {
```
Comment (alandau) [inline nifty-core/src/main/java/com/facebook/nifty/ssl/SslHandlerFactory.java]: A similar factory exists already - NiftySecurityFactory - the same one used to plug the Kerberos handlers into the pipeline. Would be nice to reuse that

Comment (siyengar) [inline nifty-core/src/main/java/com/facebook/nifty/ssl/SslHandlerFactory.java]: ya I thought about that, but that mixes with dependency inject to inject the instance. We might want to have the securityFactory and the sslhandlerfactory together for certain servers to support both security protocols. We could technically bind them with annotations, however I haven't seen a use case where SslHandlerFactory needs to be bound via DI.

File: nifty-ssl/src/main/java/com/facebook/nifty/ssl/NiftyOpenSslServerContext.java (hunk 2 above)
Comment (siyengar) [inline nifty-ssl/src/main/java/com/facebook/nifty/ssl/NiftyOpenSslServerContext.java]: will do

File: nifty-ssl/src/main/java/com/facebook/nifty/ssl/OpenSslServerConfiguration.java (hunk 1 above)
Comment (siyengar) [inline nifty-ssl/src/main/java/com/facebook/nifty/ssl/OpenSslServerConfiguration.java]: That works for me.

**Comment by siyengar** [comment]
Addressed comments.

**Pull request merged**